pkg/cli/cli.go

```diff
@@ -0,0 +1,163 @@
+package cli
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/cyberark/conjur-oss-suite-release/pkg/changelog"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/github"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/http"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/repositories"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/template"
+)
+
+// Options represents the command line values a user can pass in
+type Options struct {
+	APIToken           string
+	Date               time.Time
+	OutputFilename     string
+	OutputType         string
+	RepositoryFilename string
+	Version            string
+}
+
+type templateInfo struct {
+	TemplateName   string
+	OutputFileName string
+}
+
+const defaultOutputType = "changelog"
+const defaultRepositoryFilename = "repositories.yml"
+const defaultVersionString = "Unreleased"
+
+
+// RunParser kicks off the process for writing a new changelog
+// 1. Set the output filename if one is not set
+// 2. Collect data on each component as specified
+// 3. Build a unified changelog with each component
+// 4. Write a new changelog based on the appropriate template
+func RunParser(options Options) error {
+	templateInfo, err := getTemplateInfo(options.OutputType)
+	if err != nil {
+		return err
+	}
+	options.setOutputFilename(templateInfo.OutputFileName)
```

Comment:
This can be moved to HandleInput I think rather than being evaluated this late in the process.